### PR TITLE
Add spellcheck action

### PR DESCRIPTION
**Pull request by BeastyBlacksmith**
This will run a spellchecker over our texts and keep actions up-to date via dependabot

### CODE REVIEW

**Comment by CaptainSifff** [comment]
looks OK, do you want to fix the identified errors?

**Comment by BeastyBlacksmith** [comment]
> looks OK, do you want to fix the identified errors?

Yeah, I am going to do that, its no use to have an action that starts failing on day 1 :D

**Comment by mhagdorn** [comment]
This is really cool. It'll keep us on the straight and narrow

File: README.md [hunk 1]
```diff
@@ -7,21 +7,21 @@ Since the original idea grew in size and scope we had to split the paper in diff
 Get the latest version of the respective PDF files here:
 * "Teaching and Learning Research Software Engineering" ([intro.pdf](https://github.com/CaptainSifff/paper_teaching-learning-RSE/blob/build/intro.pdf))
 * "Foundational competencies and responsibilities of an RSE" ([competencies.pdf](https://github.com/CaptainSifff/paper_teaching-learning-RSE/blob/build/competencies.pdf))
-* "Institutionalized Organization of RSE Education" ([institutionalized_education.pdf](https://github.com/CaptainSifff/paper_teaching-learning-RSE/blob/build/institutionalized_education.pdf))
+* "Institutionalized Organisation of RSE Education" ([institutionalised_education.pdf](https://github.com/CaptainSifff/paper_teaching-learning-RSE/blob/build/institutionalised_education.pdf))
```
Comment (jngrad) [inline README.md:10]: consider renaming that Markdown file as well as the Makefile that produces the PDF file; this link is currently broken

Comment (jngrad) [reply README.md:10]: this is now fixed

**Review by jngrad — APPROVED** [review]

File: .github/workflows/spellcheck.yml [hunk 2]
```diff
@@ -3,8 +3,6 @@ on:
   workflow_dispatch:
   pull_request:
   push:
-    branches:
-      - main
```
Comment (BeastyBlacksmith) [inline .github/workflows/spellcheck.yml]: That causes the action to run twice on pull requests to main though

Comment (jngrad) [reply .github/workflows/spellcheck.yml]: sorry, I forgot to git add that file...

**Pull request merged**